public/app/features/alerting/unified/utils/matchers.ts

```diff
@@ -2,21 +2,29 @@ import { Matcher } from 'app/plugins/datasource/alertmanager/types';
 import { Labels } from '@grafana/data';
 import { parseMatcher } from './alertmanager';
 
-// parses comma separated matchers like "foo=bar,baz=~bad*" into SilenceMatcher[]
-export function parseQueryParamMatchers(paramValue: string): Matcher[] {
-  return paramValue
-    .split(',')
-    .filter((x) => !!x.trim())
-    .map((x) => parseMatcher(x.trim()));
+// Parses a list of entries like like "['foo=bar', 'baz=~bad*']" into SilenceMatcher[]
+export function parseQueryParamMatchers(matcherPairs: string[]): Matcher[] {
+  const parsedMatchers = matcherPairs.filter((x) => !!x.trim()).map((x) => parseMatcher(x.trim()));
+
+  // Due to migration, old alert rules might have a duplicated alertname label
+  // To handle that case want to filter out duplicates and make sure there are only unique labels
+  const uniqueMatchersMap = new Map<string, Matcher>();
+  parsedMatchers.forEach(
+    (matcher) => uniqueMatchersMap.has(matcher.name) === false && uniqueMatchersMap.set(matcher.name, matcher)
+  );
+
+  return Array.from(uniqueMatchersMap.values());
 }
 
 export const getMatcherQueryParams = (labels: Labels) => {
-  return `matchers=${encodeURIComponent(
-    Object.entries(labels)
-      .filter(([labelKey]) => !(labelKey.startsWith('__') && labelKey.endsWith('__')))
-      .map(([labelKey, labelValue]) => {
-        return `${labelKey}=${labelValue}`;
-      })
-      .join(',')
-  )}`;
+  const validMatcherLabels = Object.entries(labels).filter(
+    ([labelKey]) => !(labelKey.startsWith('__') && labelKey.endsWith('__'))
+  );
+
+  const matcherUrlParams = new URLSearchParams();
+  validMatcherLabels.forEach(([labelKey, labelValue]) =>
+    matcherUrlParams.append('matchers', `${labelKey}=${labelValue}`)
```

Comment:
nit: this should probably be the singular `matcher` instead of using the plural `matchers`